src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php

```diff
@@ -41,11 +41,14 @@ public function register_routes() {
 		// List themes global styles.
 		register_rest_route(
 			$this->namespace,
-			/**
-			 * Matches the theme's directory: `/themes/<subdirectory>/<theme>/` or `/themes/<theme>/`.
-			 * Note: `|` is an invalid character for a Windows' directory name.
-			 */
-			'/' . $this->rest_base . '/themes/(?P<stylesheet>[^\/\|]+(?:\/[^\/\|]+)?)',
+			// The route.
+			sprintf(
+				'/%s/themes/(?P<stylesheet>%s)',
+				$this->rest_base,
+				// Matches theme's directory: `/themes/<subdirectory>/<theme>/` or `/themes/<theme>/`.
+				// Excludes invalid directory name characters: `/:<>*?"|`.
+				'[^\/:<>\*\?"\|]+(?:\/[^\/:<>\*\?"\|]+)?'
```

Comment:
Before [changeset 52376](https://core.trac.wordpress.org/changeset/52376) , the regex was:
```php
'/' . $this->rest_base . '/themes/(?P<stylesheet>[^.\/]+(?:\/[^.\/]+)?)'
```

It was excluding `.` from being matched. The new regex removes the `.` and adds additional invalid directory name characters: `:`, `<`, `>`, `*`, `?`, `*`, `|`.